### PR TITLE
test: Assert `onSurface` in `Navigator`

### PR DESCRIPTION
**Pull request by andiwand**
Adds `assert`s to the `Navigator` to be sure that we are on the surface we expect to be.

This came up in https://github.com/acts-project/acts/pull/3437 and I pulled this out to test it separately.

### CODE REVIEW

File: Core/include/Acts/Surfaces/Surface.hpp [hunk 1]
```diff
@@ -252,12 +252,15 @@ class Surface : public virtual GeometryObject,
...
   /// @return boolean indication if operation was successful
-  bool isOnSurface(const GeometryContext& gctx, const Vector3& position,
-                   const Vector3& direction,
-                   const BoundaryTolerance& boundaryTolerance =
-                       BoundaryTolerance::None()) const;
+  bool isOnSurface(
+      const GeometryContext& gctx, const Vector3& position,
+      const Vector3& direction,
+      const BoundaryTolerance& boundaryTolerance = BoundaryTolerance::None(),
+      double tolerance = s_onSurfaceTolerance) const;
```
Comment (benjaminhuth) [inline Core/include/Acts/Surfaces/Surface.hpp:263]: Could be renamed to `onSurfaceTolerance`, to more clearly seperate it from the `boundaryTolerance`

**Review by benjaminhuth — DISMISSED** [review]
LGTM! Only one small comment, feel free to ignore!
(Funny, I was also about to propose a PR that adds the `onSurfaceTolerance` to the `isOnSurface` function).

File: Core/include/Acts/Surfaces/Surface.hpp (hunk 1 above)
Comment (andiwand) [reply Core/include/Acts/Surfaces/Surface.hpp:263]: I am also quite unhappy with `tolerance`. I think right not it is consistent in a way that we always use `tolerance` for surface tolerance and `boundaryTolerance` for the other. I would propose that we streamline this in a different PR all together

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [2cfa5f74720d53eb4f70246049e38231767c4c42]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42732672) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7997505)!

**Pull request merged**